packages/mui-material/src/Select/Select.js

```diff
@@ -114,7 +114,7 @@ const Select = React.forwardRef(function Select(inProps, ref) {
     ...(multiple && native && variant === 'outlined' ? { notched: true } : {}),
     ref: inputComponentRef,
     className: clsx(InputComponent.props.className, className),
-    variant,
+    ...(!input && { variant }),
```

Comment:
Do not propagate `variant` to root div in select input if custom input element is provided via Select's `input` prop. Basically, do not let variant be set like in the image below:

![Capture](https://user-images.githubusercontent.com/20900032/185380403-8dd55177-6484-446f-9d7f-06f4b4ea5551.PNG)

Reply:
Nice, it does not introduce a regression if I do this:

```js
<Select input={<OutlinedInput />}>
```

So, looks good.